backend/courses/teams/api.py

```diff
@@ -91,33 +91,43 @@ def request_to_join_team(request):
 
 @api_view(['PATCH'])
 @permission_classes([permissions.IsAuthenticated])
-def approve_join_team_request(request):
-    serializer = ApproveTeamMemberRequestSerializer(data=request.data)
+def manage_join_team_request(request):
+    from courses.models import TeamMember
+    serializer = ManageTeamMemberRequestSerializer(data=request.data)
     
     if serializer.is_valid():
         team_id = serializer.validated_data.get('team_id')
-        user_id = serializer.validated_data.get('user_id')
-
-        team_enrollment_data = get_student_enrollment_for_team(
-            team_id=team_id,
-            user_id=user_id,
+        joiner_name = serializer.validated_data.get('joiner_name')
+        approved = serializer.validated_data.get('approved')
+        request_id = request.user.id
+        
+        teamMembers = db.TeamMember.objects.filter(
+            team_id=team_id
         )
-        enrollment = team_enrollment_data.enrollment
-
-        try:
-            team_member = db.TeamMember.objects.get(
-                team_id=team_id,
-                enrollment=enrollment,
-                membership_type=db.TeamMember.MembershipType.REQUESTED_TO_JOIN,
-            )   
-        except db.TeamMember.DoesNotExist:
+        
+        
+        leader = None
+        approvedJoiner = None
+        for member in teamMembers:
+            if member.membership_type == TeamMember.MembershipType.LEADER:
+                if member.enrollment.user.id == request_id:
+                    leader = member
+                else: #If leader is not person requesting operation, throw error
+                    return Response(status=status.HTTP_401_UNAUTHORIZED)
+            elif member.membership_type == TeamMember.MembershipType.REQUESTED_TO_JOIN \
+                and member.enrollment.user.username == joiner_name:
+                    approvedJoiner = member
```

Comment:
Instead of iterating over all team members can we instead retrieve only the ones we're looking for 
```
try:
  joining_team_member= db.TeamMember.objects.get(
      team_id=team_id,
      membership_type=db.TeamMember.MembershipType.LEADER,
      enrollrollment__user__id=request_id,
  )   
except db.TeamMember.DoesNotExist:
     raise ValidationError("Requesting user is not team leader") # or return some bad response code
```

```
try:
  joining_team_member= db.TeamMember.objects.get(
      team_id=team_id,
      membership_type=db.TeamMember.MembershipType.REQUESTED_TO_JOIN,
      enrollrollment__user__username=joiner_name,
  )   
except db.TeamMember.DoesNotExist:
     raise ValidationError("Joining request does not exist") # or return some bad response code
```